enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -213,118 +163,94 @@ the underlying, associated Pod references CSI volumes using `Shares`
 - the `podInfoOnMount` field of the CSI Driver spec triggers the setting of these values; here is
 [an example from the hostpath plugin](https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/deploy/kubernetes-1.17/hostpath/csi-hostpath-driverinfo.yaml#L12)
-- with all this necessary metadata, the CSI driver then
-- reads of the `Secret` or `ConfigMap` noted in the `ProjectedResource`, storing it on disk on initial
-reference to said `Secret` or `ConfigMap`
-- creation of a tmpfs filesystem for each requesting `ServiceAcount` that contains the `Secret` / `ConfigMap` data
-- creation of k8s watches `ProjectedResources`, and for `Secret` / `ConfigMaps` in the namespaces listed in each discovered
-`ProjectedResources`
-- final returning of a volume that is based off of the `ServiceAccount` specific tmpfs for the data noted in the
-`volumeAttributes`
+
+With all this necessary metadata, the CSI driver then:
+
+- reads of the `Secret` or `ConfigMap` noted in the `ProjectedResource`, storing it on disk on initial reference to said `Secret` or `ConfigMap`.
+- creation of a tmpfs filesystem for each requesting `ServiceAcount` that contains the `Secret` / `ConfigMap` data.
+- creation of k8s watches `ProjectedResources`, and for `Secret` / `ConfigMaps` in the namespaces listed in each discovered `ProjectedResources`.
+- final returning of a volume that is based off of the `ServiceAccount` specific tmpfs for the data noted in the `volumeAttributes`.
 
 After initial set up of things, steady state CSI driver activities include:
-- as updates of the watched `Secrets` and `ConfigMaps` come in, SAR checks will be done for each of the previous requesting
-`ServiceAccounts` being tracked against the updated `Secret` and `ConfigMap`
-- for any `ServiceAccounts` that no longer have access, their tmpfs filesystems are unmounted, effectively removing their access to
-the data (though in prototyping and implementation, we'll need focused testing that this scales and mount propagation
-does not become a gating factor)
-- the SAR checks and if need be removal of access to data can facilitate cluster admin scenarios noted in the `User Stories`
-section like credential rotation.
-- But for `ServiceAccounts` that pass the SAR check, the tmpfs file with the data for the `Secret` / `ConfigMap` is updated
 
+- as updates of the watched `Secrets` and `ConfigMaps` come in, SAR checks will be done for each of the previous requesting `ServiceAccounts` being tracked against the updated `Secret` and `ConfigMap`.
+- for any `ServiceAccounts` that no longer have access, their tmpfs filesystems are unmounted, effectively removing their access to the data (though in prototyping and implementation, we'll need focused testing that this scales and mount propagation does not become a gating factor).
```

Comment:
I don't think you can unmount a filesystem from under a running pod (you may get `EBUSY`). Clearing the tmpfs (`rm -rf`) would be probably better.

Reply:
Yep this needs to be corrected @adambkaplan 

And yep @jsafrane that is in fact what we are doing.  

- SAR / permission check:  https://github.com/openshift/csi-driver-projected-resource/blob/master/pkg/hostpath/hostpath.go#L302-L316
- which translates to https://github.com/openshift/csi-driver-projected-resource/blob/master/pkg/hostpath/hostpath.go#L344-L356
- and the `rm -rf` equivalent at https://github.com/openshift/csi-driver-projected-resource/blob/master/pkg/hostpath/hostpath.go#L225-L239